File: Examen/Java/AlDerechoYAlReves.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Other/File.java to edit this template
 */

/**
 *
 * @author gyael
 */
import java.util.Scanner;

public class AlDerechoYAlReves {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese un numero o palabra: ");
        String cadena = sc.next();

        boolean esPalindromo = true;
        int len = cadena.length();

        for(int i = 0; i < len/2; i++) {
            if(cadena.charAt(i) != cadena.charAt(len-i-1)) {
                esPalindromo = false;
                break;
            }
        }

        if(esPalindromo)
            System.out.println("Se puede leer al derecho y al revés.");
        else
            System.out.println("No se puede.");

        sc.close();
    }
}

File: Examen/C/AlDerechoYAlreves.c
#include <stdio.h>
#include <string.h>

int main() {
    char cadena[100];
    int esPalindromo = 1;
    int i, len;

    printf("Ingrese un numero o palabra: ");
    scanf("%s", cadena);

    len = strlen(cadena);

    for(i = 0; i < len/2; i++) {
        if(cadena[i] != cadena[len-i-1]) {
            esPalindromo = 0;
            break;
        }
    }

    if(esPalindromo)
        printf("Se puede leer al derecho y al revés.\n");
    else
        printf("No se puede.\n");

    return 0;
}
